src/lock_file/satisfiability.rs

```diff
@@ -365,7 +365,8 @@ pub fn verify_package_platform_satisfiability(
     // Determine the dependencies requested by the environment
     let conda_specs = environment
         .dependencies(None, Some(platform))
-        .into_match_specs()
+        .into_specs()
+        .map(move |(name, spec)| MatchSpec::from_nameless(spec, Some(name.clone())))
```

Comment:
Do you think it would be possible to add a specialization for this in the `Dependencies` object? 

e.g

```rust
impl Dependencies<PackageName, NamelessMatchSpec> {
   pub fn into_match_specs() { .. }
}
```

Reply:
Sure :-)